zio-http/src/test/scala/zio/http/TestServer.scala

```diff
@@ -0,0 +1,155 @@
+package zio.http
+
+import zio._
+import zio.http.Server.ErrorCallback
+
+/**
+ * Enables tests that make calls against "localhost" with user-specified
+ * Behavior/Responses.
+ *
+ * @param state
+ *   State that can be consulted by our behavior PartialFunction
+ * @param behavior
+ *   Describes how the Server should behave during your test
+ * @param driver
+ *   The web driver that accepts our Server behavior
+ * @param bindPort
+ *   Port for HTTP interactions
+ * @tparam State
+ *   The type of state that will be mutated
+ */
+final case class TestServer[State](
```

Comment:
I am not sure if I would bake `State` in, as opposed to allowing the handlers to use ZIO effects.

If the handlers can return ZIO effects (containing responses), then they can already be stateful, without complicating the design of `TestServer`.

Reply:
Removed. Unsurprisingly - things are much simpler now!